General cleanup on the language specification document (#1203)
NVIDIA/cuda-quantum: docs/sphinx/specification/cudaq.rst

Language Specification 
***********************************

NVIDIA CUDA Quantum is a single-source programming model in C++ for heterogeneous 
quantum-classical computing. As such, the model provides primitive types, concepts, syntax, and
semantics that enable the integration of quantum processing units (QPUs) with
existing NVIDIA multi-node, multi-GPU parallel programming models and architectures. 

> The language is designed for C++ with extensions to other languages provided as via appropriate bindings. 

Reviewer: ```suggestion
The language is designed for C++. Extensions to other languages is fully supported via appropriate bindings. 
```

CUDA Quantum implementations are intended to be enabled via library-based language 
extensions that define quantum code as callables in the classical language being extended. 
These callables are meant to be compiled with standard compilers and runtimes 
and run via quantum computer simulators or available physical quantum architectures. The 

> language callables intended for quantum coprocessing must be annotated in some way for 

Reviewer: "in some way" seems a little ambiguous. Can we just say they must be annotated as QPU callables to be more precise here?
